### ISSUE TITLE
Network Search is not working properly/Speed Issue when there are thousands of users

### ISSUE THREAD

**Issue by ayron-paul** [buddyboss/buddyboss-platform#2400]
**Describe the bug**
The network search is not working properly when there are thousands of users. For some servers like WP Engine, the SQL query is being killed because it contains so many characters.

**To Reproduce**
1. Make sure to have thousands of users.
2. Go to BuddyBoss > Settings > Search > Select all the fields related to Members
3. On the frontend, try to search a user
4. Notice that it only shows the count of the result but it is blank

**Screenshots**
![ss1](https://user-images.githubusercontent.com/58522498/118084693-b7fa6f80-b3f3-11eb-9c67-898ba702d555.jpg)
![ss2](https://user-images.githubusercontent.com/58522498/118084695-b8930600-b3f3-11eb-93d3-ea68372608da.jpg)

**Support ticket links**
https://secure.helpscout.net/conversation/1500340606/139883/

**Comment by buntybuddyboss** [comment]
One of our client reported the speed issue for the search when there is a lot of data on the site.

Here is the explainer video: https://www.loom.com/share/87e63dcbbc9f4e19b764e658181c15da
Originally reported: https://www.dropbox.com/s/gb8gc1pse8auiav/Screen%20Recording%202021-05-21%20at%202.45.09%20PM.mov?dl=0 ( Ignore the search result ordering. )

**Comment by krunal-buddyboss** [comment]
@sourabhmatolia @firozwp 
While investigating on this issue, I found that search queries do not take that much time but some other background processes take time to load the page i.e queries on user meta table. 
Also found that if there are thousands of users the whole site become slow not only search functionality, means we need to find some thing that impacting commonly and can we optimize commonly.
So I am still working and investigate on this and we can not complete this issue on current sprint. Please apply changes according in sprint if needed.
Thanks.
FYI @chetansatasiya @KartikSuthar

**Comment by krunal-buddyboss** [comment]
@ayron-paul @buntybuddyboss 
As we have discussed and we can not reproduce this issue on server with 2lakh users with our latest platform and theme version, I am going to close this issue.
Another issue of the big SQL queries that is a one part of optimization, instead of use sub-query we have implemented two different query to make execution fast. As we have checked these both query returns the result in mile seconds only.
Thanks.
FYI @sourabhmatolia @KartikSuthar @chetansatasiya
